sea-orm-macros/src/derives/value_type.rs

```diff
@@ -0,0 +1,86 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::Type;
+
+struct DeriveValueType {
+    name: syn::Ident,
+    ty: Type,
+}
+
+impl DeriveValueType {
+    pub fn new(input: syn::DeriveInput) -> Result<Self, syn::Error> {
+        let dat = input.data;
+        let fields: Option<syn::punctuated::Punctuated<syn::Field, syn::token::Comma>> = match dat {
+            syn::Data::Struct(syn::DataStruct {
+                fields: syn::Fields::Unnamed(syn::FieldsUnnamed { unnamed, .. }),
+                ..
+            }) => Some(unnamed),
+            _ => None,
+        };
+        if fields.clone().expect("hello").into_iter().count() != 1 {
+            panic!()
+        };
```

Comment:
Hey @darkmmon, maybe we can include a friendly notice that we expect this derive macro to be used by a tuple struct which has only 1 value in the tuple? i.e. `struct Abc(T)`

> Btw... the `.expect("hello")` looks like a debug message to be, consider changing that as well?